### PR TITLE
Discuss type hint annotations in testing guide

### PR DESCRIPTION
**Pull request by namurphy**
## Description

This PR adds some discussion of type hint annotations used in tests, including the `-> None` return annotation that is used by mypy to verify that there are no return statements.  

I'm using Python 3.10+ style annotations since we'll be dropping Python 3.9 support as per NEP soon after our next release.

## Motivation and context

One of the things we'll need to do if we implement mypy in our CI (#2432) is discuss it in the contributor guide.  I'm starting with the testing guide.  I expect that type hint annotations will make our tests a lot more maintainable, especially in cases where we use `@pytest.mark.parametrize` with `args` and `kwargs`.

## Related issues

This PR came about because #2439 is adding a lot of `-> None` return annotations to tests.

### CODE REVIEW

File: docs/contributing/testing_guide.rst [hunk 1]
```diff
@@ -157,14 +157,26 @@ message to help us find the cause of a particular test failure.
 .. code-block:: python
 
    def test_addition():
-       result = 2 + 2
+       actual = 2 + 2
        expected = 4
-       assert result == expected, f"2 + 2 returns {result} instead of {expected}."
+       assert actual == expected, f"2 + 2 returns {actual} instead of {expected}."
```
Comment (namurphy) [inline docs/contributing/testing_guide.rst:162]: Unrelated change to match terminology that we often use in tests.

File: docs/contributing/testing_guide.rst [hunk 2]
```diff
@@ -157,14 +157,26 @@ message to help us find the cause of a particular test failure.
...
 .. tip::
 
    Use `f-strings`_ to improve error message readability.
 
+Type hint annotations
+---------------------
+
+PlasmaPy has begun using |type hint annotations| and |mypy| to perform
+|static type checking|. Adding a :py:`-> None` return annotation lets
+|mypy| verify that tests do not have :py:`return` statements.
```
Comment (namurphy) [inline docs/contributing/testing_guide.rst:173]:
```suggestion
PlasmaPy has begun using |mypy| to perform |static type checking| on 
|type hint annotations|. Adding a :py:`-> None` return annotation lets
|mypy| verify that tests do not have :py:`return` statements.
```

**Review by ejohnson-96 — APPROVED** [review]
Pre-commit is having an issue with whitespace in "docs/contributing/testing_guide.rst", apart from that all looks good 👍

**Comment by namurphy** [comment]
pre-commit.ci autofix
